data/moves.js

```diff
@@ -8390,16 +8390,31 @@ let BattleMovedex = {
 		selfdestruct: "ifHit",
 		slotCondition: 'healingwish',
 		effect: {
-			duration: 2,
 			onSwitchInPriority: 1,
 			onSwitchIn(target) {
-				if (!target.fainted) {
+				if (!target.fainted && (target.hp < target.maxhp || target.status)) {
+					target.heal(target.maxhp);
+					target.setStatus('');
+					this.add('-heal', target, target.getHealth, '[from] move: Healing Wish');
+					target.side.removeSlotCondition(target, 'healingwish');
+				}
+			},
+			onSwap(target) {
```

Comment:
We might consider running `Swap` after (or before, as applicable) `SwitchIn`, so this doesn't need to be duplicated.

Reply:
Running it after won't work, then it would trigger after hazards and other SwitchIn events that it should go before. Before should work, unless I've missed some switch in event that healing wish goes after.